ee/tabby-ui/app/files/components/source-code-browser.tsx

```diff
@@ -476,6 +461,49 @@ async function initFileMap(path?: string) {
   }
 }
 
+async function getFileViewType(
+  path: string,
+  blob: Blob | undefined
+): Promise<FileDisplayType> {
+  if (!blob) return ''
+  const mimeType = blob?.type
+  const detectedLanguage = filename2prism(path)?.[0]
+
+  if (mimeType?.startsWith('image')) return 'image'
+  if (detectedLanguage || mimeType?.startsWith('text')) return 'text'
+
+  const isReadableText = await isReadableTextFile(blob)
```

Comment:
@wsxiaoys  If it cannot be determined whether it's text by mimeType and filename2prism, try reading a portion of the blob to decide if it can be displayed as text.